api/apps/articles/model/index.js

```diff
@@ -0,0 +1,203 @@
+    if (err) { return callback(err, []) }
+    if (articles.length === 0) { return callback(null, []) }
+    async.map(articles, (article, cb) => {
+      article = _.extend(article, {
+        published: true,
+        published_at: moment(article.scheduled_publish_at).toDate(),
+        scheduled_publish_at: null
+      })
+      return onPublish(article, sanitizeAndSave(cb))
+    }
+    , (err, results) => {
+      if (err) { return callback(err, []) }
+      return callback(null, results)
+    })
+  })
+}
+
+export const unqueue = callback => {
+  db.articles.find({ $or: [ { weekly_email: true }, { daily_email: true } ] }, (err, articles) => {
+    if (err) { return callback(err, []) }
+    if (articles.length === 0) { return callback(null, []) }
+    async.map(articles, (article, cb) => {
+      article = _.extend(article, {
+        weekly_email: false,
+        daily_email: false
+      })
+      return onPublish(article, sanitizeAndSave(cb))
+    }
+    , (err, results) => {
+      if (err) { return callback(err, []) }
+      return callback(null, results)
+    })
+  })
+}
+
+//
+// Destroy
+//
+export const destroy = (id, callback) => {
+  find(id, (err, article) => {
+    if (err) { return callback(err) }
+    if (!article) { return callback(new Error('Article not found.')) }
+    deleteArticleFromSailthru(_.last(article.slugs), () => {
+      db.articles.remove({ _id: ObjectId(id) }, (err, res) => {
+        if (err) { return callback(err) }
+        removeFromSearch(id.toString())
+        return callback(null)
+      })
+    })
+  })
+}
+
+//
+// JSON views
+//
+export const presentCollection = async (articles) => {
+  const results = await Promise.all(_.map(articles.results, async (article) => {
+    const data = await present(article)
+    return data
+  }))
+  return {
+    total: articles.total,
+    count: articles.count,
+    results
+  }
+}
+
+export const present = async (article) => {
+  if (!article) { return {} }
+  const id = article._id ? article._id.toString() : null
+  const scheduled_publish_at = article.scheduled_publish_at ? moment(article.scheduled_publish_at).toISOString() : null
+  const published_at = article.published_at ? moment(article.published_at).toISOString() : null
+  const updated_at = article.updated_at ? moment(article.updated_at).toISOString() : null
+  const superArticleCount = await getSuperArticleCount(id)
+  return _.extend(article, {
+    id,
+    _id: id,
+    slug: _.last(article.slugs),
+    slugs: undefined,
+    published_at,
+    scheduled_publish_at,
+    updated_at,
+    is_super_sub_article: superArticleCount > 0
+  })
+}
+
+const getSuperArticleCount = (id) => {
+  return new Promise((resolve, reject) => {
+    if (!ObjectId.isValid(id)) { return resolve(0) }
+    db.articles.find({ 'super_article.related_articles': ObjectId(id) }).count((err, count) => {
+      if (err) { return reject(err) }
+      resolve(count)
+    })
+  })
+}
```

Comment:
Promisifies the mongo fetch for count.. no promise support in mongojs :(

Reply:
The official driver supports promises, and this might be [worth looking at too](https://github.com/saintedlama/mongoist).